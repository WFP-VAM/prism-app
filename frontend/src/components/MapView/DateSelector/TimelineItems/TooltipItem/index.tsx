import React, { memo } from 'react';
import { createStyles, withStyles, WithStyles } from '@material-ui/core';

const TooltipItem = memo(({ layerTitle, color, classes }: TooltipItemProps) => {
  return (
    <div className={classes.tooltipItemContainer}>
      <div>{layerTitle}</div>
      <div
        className={classes.tooltipItemColor}
        style={{
          backgroundColor: color,
        }}
      />
    </div>
  );
});

const styles = () =>
  createStyles({
    tooltipItemContainer: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
    },
    tooltipItemColor: {
      display: 'flex',
      width: 5,
      height: 5,
      marginLeft: 3,
    },
  });

interface TooltipItemProps extends WithStyles<typeof styles> {
  layerTitle: string;
  color: string;
}

export default withStyles(styles)(TooltipItem);
