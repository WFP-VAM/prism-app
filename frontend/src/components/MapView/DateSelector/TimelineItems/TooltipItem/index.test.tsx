import { render } from '@testing-library/react';
import React from 'react';
import TooltipItem from '.';

test('renders as expected', () => {
  const { container } = render(
    <TooltipItem layerTitle="Some Title" color="Some Color" />,
  );
  expect(container).toMatchSnapshot();
});
