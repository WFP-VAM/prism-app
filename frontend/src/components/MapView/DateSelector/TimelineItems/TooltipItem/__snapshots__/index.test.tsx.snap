// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="memo-tooltipItemContainer-1"
  >
    <div>
      Some Title
    </div>
    <div
      class="memo-tooltipItemColor-2"
    />
  </div>
</div>
`;
